src/catch2/internal/catch_constexpr_section.hpp

```diff
@@ -0,0 +1,58 @@
+
+//              Copyright Catch2 Authors
+// Distributed under the Boost Software License, Version 1.0.
+//   (See accompanying file LICENSE.txt or copy at
+//        https://www.boost.org/LICENSE_1_0.txt)
+
+// SPDX-License-Identifier: BSL-1.0
+#ifndef CATCH_CONSTEXPR_SECTION_HPP_INCLUDED
+#define CATCH_CONSTEXPR_SECTION_HPP_INCLUDED
+
+#if defined( CATCH_CPP17_OR_GREATER ) && __cpp_constexpr >= 201603
+#    define CATCH_CONFIG_CONSTEXPR_SECTIONS
+#endif
+
+#include <catch2/internal/catch_compiler_capabilities.hpp>
+#include <catch2/internal/catch_section.hpp>
+#include <catch2/internal/catch_unique_name.hpp>
+
+namespace Catch {
+
+    struct ConstexprSection {
+        template <typename Callable>
+        ConstexprSection( Callable callable ) {
+            static_assert(
+                [&] // error: CONSTEXPR_SECTION failure. Check the compiler
+                    // output to find the cause. You can make the section a
+                    // SECTION to investigate at runtime.
+                {
+                    callable();
+                    return true;
+                }(),
```

Comment:
```suggestion
static_assert((callable(),true));
```


I don't like lamdbas when a simple expression can do it should be simpler fpr the compiler 

if we are really pendatic we can use `void()` to always use builtin comma

Reply:
Very good suggestion, thanks. I didn't think about this, it's definitely better :+1:

Reply:
Fixed. I kept a message in the static assert to make the error clearer for the user. Even if the assert technically never fails (it's either true or unevaluable at compile time). At least compilers will spot this line and the users will see the error message in the code.

Regarding the cast to void, I considered it useless as `callable` is always a lambda that does not overload the comma operator.

Thanks for this suggestion !

Reply:
> Regarding the cast to void, I considered it useless as `callable` is always a lambda that does not overload the comma operator.

You are right